src/recipes/tests/recipes.service.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { RecipesService } from '../recipes.service';
+import { GetRecipesQueryParamsDto } from '../dto/get-recipe-query-params.dto';
+import { ForbiddenException, NotFoundException } from '@nestjs/common';
+import { mockedGetRecipesResponse } from './mocks/get-recipes-response.mock';
+import { mockedGetRecipeResponse } from './mocks/get-recipe-response.mock';
+import { GetRecipeResponse } from '../dto/get-recipe-response';
+import { mockedGetRecipesQueryParams } from './mocks/recipe-query-params.mock';
+
+describe('RecipesService', () => {
+  let service: RecipesService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        {
+          provide: RecipesService,
+          useValue: {
+            getRecipes: jest.fn(),
+            getRecipeById: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get<RecipesService>(RecipesService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+  describe('getRecipes', () => {
+    it('should call getRecipes with the correct userId and queryParams and return an array of GetRecipeResponse', async () => {
+      const userId = '1';
+      const queryParams: GetRecipesQueryParamsDto = mockedGetRecipesQueryParams;
+
+      const result = mockedGetRecipesResponse;
+      jest.spyOn(service, 'getRecipes').mockResolvedValue(result);
+
+      expect(await service.getRecipes(userId, queryParams)).toEqual(result);
+      expect(service.getRecipes).toHaveBeenCalledWith(userId, queryParams);
```

Comment:
Hint: I like to divide each test into 3 groups: given, when, then. This way more complicated tests can become more readable if know exactly what we are defining, what we are calling and what results we are expecting. The test here would look like this:

```suggestion
      // given
      const userId = '1';
      const queryParams: GetRecipesQueryParamsDto = mockedGetRecipesQueryParams;
      const mockResponse = mockedGetRecipesResponse;
      jest.spyOn(service, 'getRecipes').mockResolvedValue(mockResponse);
      
      // when
      const result = await service.getRecipes(userId, queryParams)
      
      // then
      expect(result).toEqual(mockResponse);
      expect(service.getRecipes).toHaveBeenCalledWith(userId, queryParams);
```